### PR TITLE
lib/libc : Rename LIBC_DELETE_ZONEINFO_RULES to LIBC_OPTIMIZE_ZONEINF…

### PR DESCRIPTION
**Pull request by YiHyunJin**
…O_RULES

It's not delete time zones. It just optimizes the time zone.
So I rename LIBC_DELETE_ZONEINFO_RULES to LIBC_OPTIMIZE_ZONEINFO_RULES.

### CODE REVIEW

File: lib/libc/Kconfig [hunk 1]
```diff
@@ -262,11 +262,11 @@ config LIBC_DOWNLOAD_ZONEINFO
 
 if LIBC_DOWNLOAD_ZONEINFO
 
-config LIBC_DELETE_ZONEINFO_RULES
-	bool "Delete the previous zoneinfo rules from TZ data"
+config LIBC_OPTIMIZE_ZONEINFO_RULES
+	bool "Optiminze the previous zoneinfo rules from TZ data"
```
Comment (jeongchanKim) [inline lib/libc/Kconfig]: Optiminze  -> Optimize

**Comment by seinfra** [comment]
Target : [1f6065d5dc8dfd942d2f79744cc868ed3974e204] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [1f6065d5dc8dfd942d2f79744cc868ed3974e204] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0cb711cdf0623e8971f3bb3ab2ee721f47a1c8d7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0cb711cdf0623e8971f3bb3ab2ee721f47a1c8d7] - Code Rule Check OK.

File: lib/libc/Kconfig [hunk 2]
```diff
@@ -262,12 +262,12 @@ config LIBC_DOWNLOAD_ZONEINFO
...
 
-config LIBC_DELETE_ZONEINFO_RULES
-	bool "Delete the previous zoneinfo rules from TZ data"
+config LIBC_OPTIMIZE_ZONEINFO_RULES
+	bool "Optimize zoneinfo rules from TZ data"
 	default n
 	---help---
-		Delete the previous zoneinfo rules from TZ data. Deletes the previously
-		registered zoneinfo rules at the time it is bulided.
+		Optimize zoneinfo rules from TZ data. Optimize the previously
```
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply lib/libc/Kconfig]: Delete the previously registered zoneinfo rules at the time it is built.
This sentence looks strange. All of rules were registered before built.
You want to say that `delete the previous rules unused except the latest`

**Comment by seinfra** [comment]
Target : [875b250cee57162c07c433666816be2dedabe901] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [875b250cee57162c07c433666816be2dedabe901] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [31add7fde9beb46466ca1771d8aab45b550bf029] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [31add7fde9beb46466ca1771d8aab45b550bf029] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [673a495eaa6e42f4417f0b2db3dd5c6cdd54f44f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [673a495eaa6e42f4417f0b2db3dd5c6cdd54f44f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [76fab41c92b3b4f7c3bd8629ec23d3cc7d10e420] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [76fab41c92b3b4f7c3bd8629ec23d3cc7d10e420] - Code Rule Check OK.

**Pull request merged**